message-markdown.md

```diff
@@ -0,0 +1,128 @@
+# Markdown in messages
+
+> This list is for every formating thing that goes beyond plain-text.
+
+Deltachat is going to support the following subset of markdown:
+
+### `*italics*` and `_italics_`
+
+No whitespace as first nor as end char:
+correct:
+
+```
+*italics* test
+*italics test*
+```
+
+wrong:
+
+```
+* italics* test
+```
+
+### `**bold**` and `__bold__`
+
+No whitespace as first nor as end char: see italics examples.
+
+### `~~strikethrough~~`
+
+No whitespace as first nor as end char: see italics examples.
+
+### `https://delta.chat` - Urls
+
+Make URLs clickable.
+
+### `<http://example.org>` - Urls
+
+### `` `inline-code` ``
+
+useful to send non markdown text in your message like source code snippets.
+Should get rendered in a monospace font and with a different background.
+In contrast to bold, italics and strikethrough the content of inline-code can contain spaces at begining and ending.
+
+### ` ``` fence code block ``` `
+
+```
+Similar to `inline-code` but not inline and it may supports code highlighting.
+```
+
+` ```[lang?] [content]``` ` (square brakets symbolize being a variable and are not used when using code blocks)
+A bit modified from the common syntax to allow one liners.
+Also get displayed with an monospace font (a side effect of this is that it allows to display small ascii art).
+The code **highlighting** is **optional** as implementation (time)cost
+may not be worth the small gain.
+The `language` definition should be parsed separately and omitted in this case (for oneline code blocks, because we cant differentiate it from content in this case).
+So the `language` definition is only detected if there is an linebreak following it.
+
+see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting
+
+examples:
+
+<pre>```sh
+echo hi
+```</pre>
+<pre>```
+echo hi
+```</pre>
+<pre>```echo hi```</pre>
+
+
+### `:emoji:`
+
+- could also be used for custom dc emojis
+
+### labeled links: `[Name](url)` links
+
+When implementing this make sure to show the user the hidden url in a confirmation dialog to make scamming harder.
+Also show the url as encode punycode to make punycode attacks useless.
+Optionaly a client can implement a system to trust an domain (an "don't ask a again for links on this domain" checkbox in the confirmation dialog)
+
+### Bot `/commands`
+
+On click the command gets prefilled as the draft so it can be easialy send.
+
+## `#tag`
+
+Basically a clickable search shortcut. On click it opens the message search prefilled with that tag.
+
+Inspired by twitters and telegrams #hashtag funtionality.
```

Comment:
we still need to discuss how to parse hashtags:
telegrams rule seems to be
- `#` everything (besides `#`) until space/linebreak/tab
- must end with EOF/space/linebreak/tab (but don't consume it)